index.js

```diff
@@ -1,11 +1,47 @@
-const request = require("request");
+const rp = require('request-promise')
+const retries = process.env.RETRIES || 3
+const delay = process.env.RETRY_DELAY || 1000
+const timeout = process.env.TIMEOUT || 5000
+
+const requestRetry = (options, retries) => {
+    return new Promise((resolve, reject) => {
+        const retry = (options, n) => {
+            return rp(options)
+                .then(response => {
+                    if (response.body.error) {
+                        if (n === 1) {
```

Comment:
```suggestion
                        if (n < 1) {
```

`<` because it's safer and `1` because otherwise it would only retry twice, not three times

Reply:
This is actually working as intended. There will be a maximum total of 3 attempts to reach the endpoint (initial attempt + 2 retries). We are following the common naming convention for frameworks like this, by defining the amount of retries to "3".

I have changed this to less than or equal to 1. Does this work for you?

Reply:
Personally, I'd say the variable should be `tries` instead of `retries` then, but that's cherrypicking. Works for me :-)